c3d/c3d.py

```diff
@@ -1105,17 +1236,19 @@ def add_group(self, group_id, name, desc):
         ------
         TypeError
             Input arguments are of the wrong type.
+        KeyError
+            Name or numerical key already exist (attempt to overwrite existing data).
         '''
         if not is_integer(group_id):
-            raise ValueError('Expected Group numerical key to be integer, was %s.' % type(group_id))
+            raise TypeError('Expected Group numerical key to be integer, was %s.' % type(group_id))
         if not (isinstance(name, str) or name is None):
-            raise ValueError('Expected Group name key to be string, was %s.' % type(name))
-        group_id = int(group_id) # Assert python int
+            raise TypeError('Expected Group name key to be string, was %s.' % type(name))
+        group_id = int(group_id) # Asserts python int
         if group_id in self._groups:
-            raise KeyError(group_id)
+            raise KeyError('Group with numerical key {} already exists'.format(group_id))
         name = name.upper()
         if name in self._groups:
-            raise KeyError(name)
+            raise KeyError('No group matched name key {}'.format(name))
```

Comment:
I don;'t think the text here fits. I would prefer the message I added in #45

Reply:
But I guess this entire function will be changed once #45 is merged

Reply:
Yeah saw that the message was not descriptive at all when I checked the line for the other pull request, so good find! Removed the change to avoid conflict, the error message two rows above it is accurate though and should have been used for the name key as well!